service/history/queues/executable.go

```diff
@@ -353,7 +357,7 @@ func (e *executableImpl) HandleErr(err error) (retErr error) {
 	}
 	e.resourceExhaustedCount = 0
 
-	if _, isNotFound := err.(*serviceerror.NotFound); isNotFound {
+	if common.IsNotFoundError(err) {
```

Comment:
🙏  I hope we don't wrap notFound errors in cases where tasks need to be retried...

Reply:
I can revert this change if you'd like